### PR TITLE
docs(readme): update README to include the minimum node.js version

### PR DESCRIPTION
**Pull request by lapa182**
On the previous PR, we missed updating the README to reflect the new Node.js engine.

➕ Extra change:
We were [rate limited](https://github.com/duffelhq/duffel-api-javascript/actions/runs/4881478277/jobs/8710431028) on GitHub API which caused our workflow to fail, this is due to an issue with semantic-release. I've disabled comments on issues to prevent this from happening.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -101,7 +101,13 @@
...
       ],
       "@semantic-release/npm",
-      "@semantic-release/github"
+      [
+        "@semantic-release/github",
+        {
+          "successComment": false,
+          "failTitle": false
+        }
+      ]
```
Comment (shaundon) [inline package.json:110]: This seems unrelated – what's the background here?

Comment (lapa182) [reply package.json:110]: Yeah, sorry I should have mentioned this on the description (which I will add). This is because of this: https://github.com/duffelhq/duffel-api-javascript/actions/runs/4881478277/jobs/8710431028

Comment (shaundon) [reply package.json:110]: Ah thanks for the context. I'm still a bit confused though – the thing you linked to mentions that we've hit a rate limiting issue?

Comment (lapa182) [reply package.json:110]: Yes, indeed. This is because every time we publish a new release we comment on GH issues which causes our workflow to fail. Please see https://github.com/semantic-release/semantic-release/issues/2204 for further informations on the work-around and issue

Comment (shaundon) [reply package.json:110]: Ah I see – thanks! It's a shame we can't add comments to package.json to give more context, but we can always check the git blame for it in the future and find this PR at least

**Review by shaundon — APPROVED** [review]
